src/timepicker/timepicker.js

```diff
@@ -159,10 +173,14 @@ class TimePicker extends React.Component<TimePickerPropsT, TimePickerStateT> {
   };
 
   handleChange = (seconds: number) => {
-    const date = this.props.value ? new Date(this.props.value) : new Date();
-    const [hours, minutes] = secondsToHourMinute(seconds);
-    date.setHours(hours, minutes, 0);
-    this.props.onChange && this.props.onChange(date);
+    const {utils} = this.context;
+    const {setHours, setMinutes, setSeconds} = utils;
+    const date = utils.date(this.props.value || undefined);
+    const [hours, minutes] = secondsToHourMinute(seconds, utils);
+    const hourDate = setHours(date, hours);
+    const minuteDate = setMinutes(hourDate, minutes);
+    const updatedDate = setSeconds(minuteDate, 0);
```

Comment:
```suggestion
    const date = utils.date(this.props.value || undefined);
    const [hours, minutes] = secondsToHourMinute(seconds, utils);
    const hourDate = utils.setHours(date, hours);
    const minuteDate = utils.setMinutes(hourDate, minutes);
    const updatedDate = utils.setSeconds(minuteDate, 0);
```

nit: I find it a bit tough to follow the multiple destructures (was unsure for a moment where the functions were coming from). No need to commit, just a suggestion